editor.planx.uk/src/components/Footer.tsx

```diff
@@ -42,49 +42,85 @@ export interface Props {
   children?: React.ReactNode;
 }
 
+// XXX: This component is located in the DOM within the Footer component, but as it's appearance is controlled by a global window "message" event. This means that its appearance can be controlled by other instances of the FeedbackFish widget (such as in the PhaseBanner)
+const FeedbackPrivacyNote = ({
+  onClose,
+}: {
+  onClose: (e: React.MouseEvent) => void;
+}) => (
+  <Dialog
+    // XXX: only render dialog when visible & open=true by default,
+    //      otherwise clicking modal BG also closes feedback widget
+    open
+    onClose={onClose}
+    // feedback fish uses z-index of 999999 :(
+    style={{ zIndex: 999999 + 1 }}
+    onClick={(e) => e.stopPropagation()}
+  >
+    <DialogContent>
+      Please do not include any personal or financial information in your
+      feedback
+    </DialogContent>
+    <DialogActions>
+      <Button onClick={onClose}>OK</Button>
+    </DialogActions>
+  </Dialog>
+);
+
 export default function Footer(props: Props) {
   const { items, children } = props;
   const [feedbackPrivacyNoteVisible, setFeedbackPrivacyNoteVisible] =
     useState(false);
+  const [metadata, setMetadata] = useState<Record<string, string>>();
+
+  const feedbackFishId = process.env.REACT_APP_FEEDBACK_FISH_ID;
 
   useEffect(() => {
     let feedbackFishPostMessageWorkingCorrectly: boolean;
     const handleMessage = (event: MessageEvent) => {
-      try {
-        // the feedback fish widget posts a message that's either
-        // {"width":0,"height":0} if the iframe is hidden
-        // or {"width":320,"height":200} if the iframe is visible
-        if (event.origin.endsWith("feedback.fish")) {
-          const { width, height } = JSON.parse(event.data);
-          // XXX: postMessage not working in firefox as expected
-          //      https://trello.com/c/MX1cpAM8 this disables it in FF
-          //      by checking if the initial {"width":0,"height":0} message
-          //      was received. If not then never show feedback warning msg
-          if (width === 0 && height === 0) {
-            feedbackFishPostMessageWorkingCorrectly = true;
-          }
-          setFeedbackPrivacyNoteVisible(
-            width > 0 && height > 0 && feedbackFishPostMessageWorkingCorrectly
-          );
-        }
-      } catch (err) {}
+      setMetadata(getFeedbackMetadata());
```

Comment:
This is the only new logic in this component, everything else is just splitting up / tidying up / adding comments.